### DISCUSSION TITLE
Autocompletion & type hinting with custom tenant model

### DISCUSSION THREAD

**Discussion by vpratfr** [spatie/laravel-multitenancy#313] [Ideas]
Hi,

I have integrated recently your package in our project, and one thing that was bugging me was the lack of proper type hints when doing things as `Tenant::current()`

Then I read more about Generics in PHPDoc that could help me somewhere else in my code, and realized that could be in fact a possible way to get type hinting with custom models.

Here is a link to the article I read: https://phpstan.org/blog/generics-in-php-using-phpdocs

I will have a look if I get some time to submit a PR if you have not looked into it already and if you are interested in that. Should not break anything as this would only add comments.

**Comment by masterix21** [comment]
Hi @vpratfr.

https://github.com/spatie/laravel-multitenancy/blob/804ad0f1e41a5d6d9d306d383b08a054e790d9de/src/Models/Tenant.php#L45

`current()` already returns the type.

Using a class like so `class YourCustomTenant extends Tenant` and calling `YourCustomTenant::current()`, your type-hint should return the right tenant model. Isn't right?

**Comment by vpratfr** [comment]
Current not, sorry, or at least not on the latest PHPStorm release

![image](https://user-images.githubusercontent.com/2526465/146363483-f1dd32fc-049e-48e6-9ddc-0396059a1e01.png)

I guess that would work with the `static` return type or by specifying that in a phpdoc comment.
